src/mem.rs

```diff
@@ -23,6 +23,26 @@ impl<T> MemPool<T> {
         Vec::with_capacity(size)
     }
 
+    /// A version of [`Self::pop`] that initializes the [`Vec`].
+    /// This allows it to use [`vec!`], which, if used with `0`,
+    /// calls [`alloc_zeroed`], and thus can leave zero initialization to the OS.
+    ///
+    /// [`alloc_zeroed`]: std::alloc::alloc_zeroed
+    pub fn pop_init(&self, size: usize, init_value: T) -> Vec<T>
```

Comment:
If the point of this function is to allow for zero-initialization, does it make sense to have the caller specify the init value? It seems like it'd be better to always zero-initialize.

Reply:
What about `T: Copy + Default` and we use `buf.resize(size, Default::default())`?

Reply:
> What about `T: Copy + Default` and we use `buf.resize(size, Default::default())`?

`buf.resize` doesn't have the optimization, `Vec::from_elem` (i.e. `vec!`) does.

> If the point of this function is to allow for zero-initialization, does it make sense to have the caller specify the init value? It seems like it'd be better to always zero-initialize.

It's easier this way with `MemPool` being generic.  I could also do something like `pop_init_default` and use `Default::default()`, but I don't really see much benefit in doing that instead (it just constrains the API more).